napari/utils/colormaps/colormap.py

```diff
@@ -162,60 +162,138 @@ class LabelColormap(Colormap):
-        mapped = self.colors[values.astype(np.int64)]
+        return _cast_labels_data_to_texture_dtype(
+            np.array([self.selection]).astype(dtype), self
+        )[0]
 
-        mapped[values == self.background_value] = 0
+    def background_as_minimum_dtype(self, dtype: np.dtype) -> int:
+        """Treat background as given dtype and calculate value with min dtype.
 
-        # If using selected, disable all others
-        if self.use_selection:
-            cast_selection = _cast_labels_to_minimum_dtype_auto(
-                np.array([self.selection]),
-                len(self.colors) - 1,
-                self.background_value,
-            )[0]
-            mapped[values != cast_selection] = 0
+        Parameters
+        ----------
+        dtype : np.dtype
+            The dtype to convert the background to.
 
-        return mapped
+        Returns
+        -------
+        int
+            The background converted.
+        """
+        return _cast_labels_data_to_texture_dtype(
+            np.array([self.background_value]).astype(dtype), self
+        )[0]
+
+    def map(self, values, apply_selection=True) -> np.ndarray:
+        """Map values to colors.
+
+        Parameters
+        ----------
+        values : np.ndarray or float
+            Values to be mapped.
+        apply_selection : bool
+            Whether to apply selection if self.use_selection is True.
+
+        Returns
+        -------
+        np.ndarray of same shape as values, but with last dimension of size 4
+            Mapped colors.
+        """
+        original_shape = np.shape(values)
+        values = np.atleast_1d(values)
+
+        if values.dtype.kind == 'f':
+            values = values.astype(np.int64)
+
+        if values.dtype == np.uint8 and "_uint8_colors" in self.__dict__:
+            # __dict__ checks whether _uint8_colors is cached — if not, it
+            # falls back on else to map all the colors, avoiding an infinite
+            # recursion.
+            mapped = self._uint8_colors[values]
+        elif values.dtype == np.uint16 and "_uint16_colors" in self.__dict__:
+            # same as above uint8 clause.
+            mapped = self._uint16_colors[values]
+        else:
+            background = self.background_as_type(values.dtype)
+            # cast background to values dtype to support int8 and int16
+            # negative backgrounds
+            texture_dtype_values = _zero_preserving_modulo_numpy(
+                values, len(self.colors) - 1, values.dtype, background
+            )
+            mapped = self.colors[texture_dtype_values]
+            mapped[texture_dtype_values == 0] = 0
+        if self.use_selection and apply_selection:
+            selection = self.selection_as_type(values.dtype)
+            # cast selection to values dtype to support int8 and int16
+            # negative backgrounds
+            mapped[(values != selection)] = 0
+
+        return np.reshape(mapped, original_shape + (4,))
```

Comment:
@jni Why it is important? (it crashes tests).

Reply:
We can remove — but then the docstring is lying. imho it's the more consistent behaviour, though, so it would be good to keep it and fix tests. Sorry, I thought I ran all the relevant tests but clearly not.

Reply:
I prefer to removie it from here and do in #6439 for both modes. It should simplify fixing test also.